src/shared/gitlab-client-spec.ts

```diff
@@ -114,4 +102,63 @@ describe('gitlab-client', () => {
     });
   });
 
+  describe('getUserByEmail', () => {
+    it('returns a single user', async () => {
+      // Arrange
+      const gitlab = getClient();
+      fetchMock.restore();
+      const email = 'foo@bar.com';
+      fetchMock.mock(/\/users/, [{
+        id: 1,
+        email,
+      }]);
+
+      // Act
+      const response = await gitlab.getUserByEmail(email);
+
+      // Assert
+      expect(response.id).to.equal(1);
+
+    });
+    it('throws when not found', async () => {
+      // Arrange
+      const gitlab = getClient();
+      fetchMock.restore();
+      const email = 'foo@bar.com';
+      fetchMock.mock(/\/users/, []);
+
+      // Act
+      try {
+        await gitlab.getUserByEmail(email);
+      } catch (err) {
+        // Assert
+        expect(err).to.exist;
+        return;
+      }
+      expect(false, 'Shouldn\'t get here').to.be.true;
+
+    });
+  });
+
+  describe('getUserTeams', () => {
+    it('returns an array of teams', async () => {
```

Comment:
Should there also be a negative test, i.e. it doesn't return a team for either an invalid user or a user that's not on any teams?